src/NuGet.Core/NuGet.Packaging/PackageArchiveReader.cs

```diff
@@ -173,7 +173,14 @@ protected override void Dispose(bool disposing)
                         var attr = File.GetAttributes(copiedFile);
                         if (!attr.HasFlag(FileAttributes.Directory))
                         {
-                            File.SetLastWriteTimeUtc(copiedFile, entry.LastWriteTime.UtcDateTime);
+                            try
+                            {
+                                File.SetLastWriteTimeUtc(copiedFile, entry.LastWriteTime.UtcDateTime);
+                            }
+                            catch (ArgumentOutOfRangeException)
```

Comment:
I was thinking of just checking if the entry's LastWriteTime is in the future, if so then ignore it. But many times in the future do work. After this fix is done I can come up with a minimal mono repro and log a bug.